src/components/dashjs-api-control/dashjs-api-control.tsx

```diff
@@ -239,9 +251,24 @@ export class DashjsApiControl {
       return;
     }
     if (this.selectedFunctions.has(key)) {
-      this.selectedFunctions.set(key, []);
+      const currFunction = this.functionList.filter(s => s.name === key)[0];
```

Comment:
this function differs from the one in the modal, is this on purpose? Might be good to use the same ( `utils`?)

Reply:
i mean the foreach switch below